### PR TITLE
Fix bug that can't parse materials with implicit library specification. 

### LINKED ISSUE #365: Material Specifications: cannot unpack non-iterable ValueNode object
@MicahGale How are material cards supposed to be formatted for MontePy to parse them? I keep getting the error in the subject line. When I inspect the `isotope_fractions` variable in `\data_inputs\material.py`, it is a `ListNode` object containing `ValueNode` objects. For any `ValueNode` object, I see no attribute that corresponds to the isotopic fraction (at least not one intuitively named). Additionally this object is non-iterable so the usage isn't correct either. Am I missing something?

### PR DESCRIPTION
**Pull request by MicahGale**
The material parser was built assuming that all isotope specifications required a library, e.g., `1001.80c`. However according to the 6.2 manual table 3-32: 

> Either a full ZZZAAA.abx or partial ZZZAAA element or nuclide
identifier for each constituent i, where ...

Therefore, this assumption is flawed.

To fix this, a dedicated `MaterialParser` was created. `1001` is ambiguous if it is part of a number sequence or a ZAID. So this parser just accepts both Isotopes and number sequences as valid. This avoids the need of contextual parser with a look-ahead capability by making the input contextual as an input level. 

@SterlingButters you can review this if you like, but you don't need to. 

As for the specific error message that was found in #365. The Material assumed it had an `IsotopesNode` which is a list of tuples, and was given a `ListNode` instead. It tried to unpack the `ValueNode` inside the list like a tuple, and freaked out. 

Fixes #365.

### CODE REVIEW

**Comment by MicahGale** [comment]
This needs to wait until #336 is merged due to the de facto deployment freeze depending on that.

**Review by tjlaboss — APPROVED** [review]

**Comment by MicahGale** [comment]
Thank you!

**Pull request merged**
